README.md

```diff
@@ -4,40 +4,128 @@
+```
+
+To facilitate this simple workflow, there are some assumed data for each component to achieve a successfull publish:
+
+
+```python
+{
+  "assettype_data": {
+    "short": "upload", # Default asset type is "upload".
+  },
+  "asset_data": {
+    "name": task_name, # This is the name of the current task, collected by the extension.
+    "type": assettype_entity, # The AssetType entity derived from the "assettype_data".
+    "parent": task_parent, # The parent of the current task.
+  },
+  "assetversion_data": {
+    "version": 0, # This is to ensure a version 1 gets created if non exists.
+    "asset": asset_entity, # The Asset entity derived from the "asset_data".
+    "task": task, # The current task.
+  },
+  "component_data": {
+    "name": "main", # Default component name is "main".
+    "version": assetversion_entity, # The AssetVersion entity derived from "assetversion_data".
+  },
+  "component_overwrite": False, # Overwriting will deleted existing data, if supported by the location, and component, and then create a new component.
+  "component_location": session.pick_location(), # This will pick the highest prioritized location.
+}
+```
+
+All of this data can be overwritten by providing it in the component data item in ```ftrackComponentsList```, and you can also provide more data. It is also worth noting that you can add any number of components to the ```ftrackComponentsList``` per instance at any point until the integrator executes, which could have completely different data so you can publish any component.
+Here is an example of overriding the default, and add another component of a different type:
+
+```python
+instance.data["ftrackComponentsList"] = [
+  {
+    "assettype_data": {
+      "short": "scene"
+    },
+    "assetversion_data": {
+      "version": 2
+    },
+    "component_data": {
+      "name": "mayaAscii"
+    },
+    "component_path": "path/to/file",
+    "component_overwrite": True,
+  },
+  {
+    "assettype_data": {
+      "short": "mov"
+    },
+    "assetversion_data": {
+      "version": 2
+    },
+    "component_data": {
+      "name": "movie"
+    },
+    "component_path": "path/to/movie/file",
+    "component_overwrite": True
+  },
+]
+```
+
+### Old API
 
 To succesfully publish an instance to ftrack using this extension you need to append a few data members to the instance you want to publish.
 
-**Required data:**
+**Required data**
 
-- ftrackComponents - Dictionary of dictionaries where each key is a component name (Instance data)
- - Each component (key) should be a dictionary that can have 2 keys
-   - *path* (required): holds path to the file being published
-    - *reviewable* (optional): tells extension whether you want component to be uploaded to ftrack servers and made available for webReview
+- ftrackComponents (dict) - Dictionary of dictionaries where each key is a component name. There are required and optional keys:
```

Comment:
Have a look at whether a table is a good fit for this kind of data. I find it useful for two-dimensional data like this.

[Example](https://github.com/mottosso/Qt.py#documentation)